### ISSUE TITLE
Tiddlywiki (with nodejs) crashes if we set a username and password

### ISSUE THREAD

**Issue by heartsmagic** [Jermolene/TiddlyWiki5#2497]
I don't know if this is related with TiddlyWiki or Node.js but if we set a username and password nodejs throws that error:

```
_http_outgoing.js:309
    throw new TypeError('The header content contains invalid characters');
    ^

TypeError: The header content contains invalid characters
    at storeHeader (_http_outgoing.js:309:11)
    at ServerResponse.OutgoingMessage._storeHeader (_http_outgoing.js:223:9)
    at ServerResponse.writeHead (_http_server.js:218:8)
    at Server.<anonymous> ($:/core/modules/commands/server.js:111:14)
    at emitTwo (events.js:100:13)
    at Server.emit (events.js:185:7)
    at HTTPParser.parserOnIncoming [as onIncoming] (_http_server.js:542:12)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:103:23)

```

Tiddlywiki: 5.1.12
Nodejs:  5.12.0

**Comment by heartsmagic** [comment]
I updated the issue since it occured on Chrome, too.

**Comment by sukima** [comment]
I believe this is relevant: http://www.alexkras.com/typeerror-the-header-content-contains-invalid-characters/

Looking at [the source (`server.js` line 111)](https://github.com/Jermolene/TiddlyWiki5/blob/master/core/modules/commands/server.js#L111-L113) I see the only dynamic part of this is the **server name**

@heartsmagic What is the value of your server name? I bet there are non ASCII characters in it.

@Jermolene Perhaps a `$tw.utils.stringify(…)` would help.

**Comment by heartsmagic** [comment]
@sukima, at first I thought that this could be a locale issue since not so much people encounter that problem. Also I saw that blog entry but did not understand too much. 

My server installed with my locale (Turkish), at least I remember like this but now I'm using it with en_US.UTF-8 and my server name never had non-ASCII characters. I'am using "sherlock" since the beginining. As I said beause I suspected for a locale issue, I tried to start Tiddlywiki with LC_ALL=C but that did not work and I decided to open a bug record here. 

But, since you pointed the file and relevant part I cut out the "servername" and it worked! 

Thanks, at least we are sure that the problem is this. But, what could be the problem about my server name?

**Comment by sukima** [comment]
If you like a few lines up it is getting the server name from the TiddlyWiki site title. So it is what ever you called your TiddlyWiki edition. This could have non-ASCII characters or there may be a line feed or non breaking space. Either way it is a bug that it isn't sanitized for use in HTTP headers hence my Pull Request.

If you want a work around for now I would suggest examining the TiddlyWiki edition site title (in the settings as you view it)

What does the tiddler file for `$:/SiteTitle` look like?

**Comment by heartsmagic** [comment]
So sorry for being too late but yes it is because of non-ascii characters in our SiteTitle. If we change to an ascii one problem does not occure.

**Comment by ght** [comment]
This should be fixed since 2018-08-24 in e237d8fa974b which fixed #3410 (discussion there mentions that the currently still open #1930 might be related).

The fix escapes the SiteTitle using `transliterateToSafeASCII` before assigning it to `servername` (see [`core/modules/server/server.js:35`](https://github.com/Jermolene/TiddlyWiki5/blob/b8fa6f0f0/core/modules/server/server.js#L35)):
```diff
-	this.servername = this.wiki.getTiddlerText("$:/SiteTitle") || "TiddlyWiki5";
+	this.servername = $tw.utils.transliterateToSafeASCII(this.wiki.getTiddlerText("$:/SiteTitle") || "TiddlyWiki5");
```

Stale PR #2498 which was intended as a fix for this but seems abandoned since 2018-11-27 can be closed together with this.

**Comment by Jermolene** [comment]
Thanks @ght, appreciated.
